src/MainWindow.vala

```diff
@@ -137,6 +137,26 @@ public class Maya.MainWindow : Gtk.ApplicationWindow {
         }
     }
 
+    public void on_duplicate (ECal.Component comp) {
+        E.Source src = comp.get_data ("source");
+
+        if (src.writable == true && Model.CalendarModel.get_default ().calclient_is_readonly (src) == false) {
+            // The event editor dialog (EventDialog) uses its date/time parameter to tell
+            // if we're editing an existing event (parameter is null) or creating a new one
+            // (parameter is not null). Since here we're creating a new event as a copy of
+            // an existing one, we have to pass that event's date/time.
+            DateTime from_date, _;
+            Util.get_local_datetimes_from_icalcomponent (comp.get_icalcomponent (), out from_date, out _);
+
+            // Now open the editor dialog.
+            var dialog = new Maya.View.EventDialog (comp, from_date);
```

Comment:
The dialog needs to be amended so that it can distinguish between creating an new event, editing an existing event  and duplicating an existing event.  At the moment it does not do this properly and therefore does not duplicate recurring events (just create a new standard event at the same time)